src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,400 @@
+    getValue: data => parseTimeCommitment(data.hours_per_week, data.num_weeks),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: data => (data.is_remote ? 'Remote' : 'In Person'),
+  },
+];
+
+const intepretationFields: ListingField<Interpretation>[] = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: data => data.languages.join(', '),
+  },
+  // Remote/In Person
+  {
+    label: 'Remote/In Person',
+    getValue: data => (data.is_remote ? 'Remote' : 'In Person'),
+  },
+];
+
+const docFields: ListingField<DocumentTranslation>[] = [
+  // Languages
+  {
+    label: 'Language(s)',
+    getValue: data => data.languages.join(', '),
+  },
+  // Number of Pages
+  {
+    label: 'Number of Pages',
+    getValue: data => data.num_pages.toString(),
+  },
+];
+
+const lcaFields: ListingField<LimitedCaseAssignment>[] = [
+  // Country Field
+  {
+    label: 'Country Field',
+    getValue: data => data.country,
+  },
+  // Language(s)
+  {
+    label: 'Language(s)',
+    getValue: data => data.languages.join(', '),
+  },
+  // Expected Deliverable
+  {
+    label: 'Expected Deliverable',
+    getValue: data => data.deliverable,
+  },
+];
+
+export default function ListingDetails({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
+  const auth = useAuth();
+  const profile = useProfile();
+
+  const listingFields = useMemo(() => {
+    let fields: ListingField<any>[];
+    if (listingData.listing_type === 'CASE') {
+      if (interpretation && listingData.needs_interpreter) {
+        fields = caseInterpretationFields;
+      } else {
+        fields = [
+          ...caseFields,
+          {
+            label: listingData.is_detained
+              ? 'Custody Location'
+              : 'Client Location',
+            getValue: data => data.client_location || 'N/A',
+          },
+        ];
+      }
+    } else if (listingData.listing_type === 'INT') {
+      fields = intepretationFields;
+    } else if (listingData.listing_type === 'DOC') {
+      fields = docFields;
+    } else {
+      fields = lcaFields;
+    }
+
+    return (
+      <InnerFieldContainer>
+        {fields.map(({ label, getValue }) => (
+          <div key={label}>{renderField(label, getValue(listingData))}</div>
+        ))}
+      </InnerFieldContainer>
+    );
+  }, [listingData, interpretation]);
+
+  const dateComponent = useMemo(() => {
+    if (listingData.listing_type !== 'INT') {
+      let dateHeader = 'Deadline';
+      let dateData = '';
+      if (listingData.listing_type === 'CASE') {
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date);
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendarBig" />
+          <Subheading $bold>
+            {dateHeader}: <Subheading> {dateData}</Subheading>
```

Comment:
it seems like nesting `<p>` elements is giving us this console error:
<img width="1440" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/d9086402-59d3-4574-897e-d936e7a0718e">

Let's go ahead and move the nested `Subheading` outside so that it's at the same level as the other `Subheading` and `Icon`

Reply:
gotcha!